flagsmith/flagsmith.py

```diff
@@ -190,27 +218,29 @@ def get_identity_segments(
         segment_models = get_identity_segments(self._environment, identity_model)
         return [Segment(id=sm.id, name=sm.name) for sm in segment_models]
 
-    def update_environment(self):
+    def update_environment(self) -> None:
         self._environment = self._get_environment_from_api()
 
     def _get_environment_from_api(self) -> EnvironmentModel:
         environment_data = self._get_json_response(self.environment_url, method="GET")
         return EnvironmentModel.model_validate(environment_data)
 
-    def _get_environment_flags_from_document(self) -> Flags:
-        return Flags.from_feature_state_models(
-            feature_states=engine.get_environment_feature_states(self._environment),
-            analytics_processor=self._analytics_processor,
-            default_flag_handler=self.default_flag_handler,
-        )
+    def _get_environment_flags_from_document(self) -> Flags:  # type: ignore[return]
```

Comment:
Again, this `type: ignore` should go away if we opt for a simple `None` check.

Reply:
```py
def _get_environment_flags_from_document(self) -> Flags:
    if self._environment:
        return Flags.from_feature_state_models(
            feature_states=engine.get_environment_feature_states(self._environment),
            analytics_processor=self._analytics_processor,
            default_flag_handler=self.default_flag_handler,
        )
```

Using `None` still gives a `mypy` error. I have to probably raise an error in the else clause to fix the `no return` error but that looks like additional noise.

Reply:
This is fine by me:

```python
if self._environment is None:
    raise TypeError('No environment present')
```

Reply:
Added